desc/protoparse/parser_test.go

```diff
@@ -422,3 +422,106 @@ message Foo {
 	comment := fds[0].GetMessageTypes()[0].GetFields()[0].GetSourceInfo().GetLeadingComments()
 	testutil.Eq(t, " leading comments\n", comment)
 }
+
+func TestParseInferImportPaths_SimpleNoOp(t *testing.T) {
+	sources := map[string]string{
+		"test.proto": `
+		syntax = "proto3";
+		import "google/protobuf/struct.proto";
+		message Foo {
+			string name = 1;
+			repeated uint64 refs = 2;
+			google.protobuf.Struct meta = 3;
+		}`,
+	}
+	p := Parser{
+		Accessor:         FileContentsFromMap(sources),
+		InferImportPaths: true,
+	}
+	fds, err := p.ParseFiles("test.proto")
+	testutil.Ok(t, err)
+	testutil.Eq(t, 1, len(fds))
+}
+
+func TestParseInferImportPaths_FixesNestedPaths(t *testing.T) {
```

Comment:
This more involved test actually verifies that `InferImportPaths` is working and doing what it's supposed to do.